go/pserver/cclient/cclient.go

```diff
@@ -100,9 +105,39 @@ func paddle_new_pserver_client(addrs *C.char, selected int) C.paddle_pserver_cli
 }
 
 //export paddle_new_etcd_pserver_client
-func paddle_new_etcd_pserver_client(etcd_addr *C.char) C.paddle_pserver_client {
-	// TODO(helin): fault tolerant pserver client using etcd.
-	panic("not implemented.")
+func paddle_new_etcd_pserver_client(etcd_addr *C.char, pserver_path *C.char, selected int) C.paddle_pserver_client {
+	if etcd_addr == "" {
+		return C.PSERVER_ERROR
+	}
+	etcd_addrs := strings.Split(etcd_addr, ",")
```

Comment:
Sorry I should have mentioned this earlier, the cgo code are supposed to be just a thin layer. Because it's just a interface from C to Go. If too much code here, then when we want to use Pserver from Go to Go, we need to duplicate code. Another reason is Go's test does not work well with cgo, it's supposed to be tested from C side. It's much easier to write test for Go.

You can create a new file in pserver package. E.g., etcd.go, and implement this interface: https://github.com/PaddlePaddle/Paddle/blob/develop/go/pserver/client.go#L26
And this interface:
https://github.com/PaddlePaddle/Paddle/blob/develop/go/pserver/client.go#L15

So cgo just need to do:
```Go
e := pserver.NewEtcd(addr) // we can call it NewEtcd, or some other name.
c := pserver.NewClient(e, len(as), e)
return add(c)
```